### PR TITLE
Passive scan for Java error messages containing sensitive information (CWE-209)

### PR DESCRIPTION
**Pull request by ChieftainY2k**
Passive scan for Java error messages containing sensitive information (CWE-209)

### CODE REVIEW

**Comment by thc202** [comment]
It's missing the DCO, https://github.com/zaproxy/community-scripts/pull/386/checks?check_run_id=16132764226

File: passive/JavaDisclosure.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    var alertConfidence = 3
+    var alertTitle = 'Java stack trace disclosure (or similar) - investigation required (script)'
+    var alertDesc = 'Java stack trace disclosure (or similar) was found'
+    var alertSolution = 'Investigate Java stack trace disclosures found in the response, remove or mask as required'
+    var cweId = 209
+    var wascId = 0
+
+    var re = /springframework|\.java|rootBeanClass/i
+
+    var url = msg.getRequestHeader().getURI().toString()
```
Comment (thc202) [inline passive/JavaDisclosure.js:14]: Should be done later (when raising the alert) to void creating a new string unnecessarily.

**Comment by thc202** [comment]
The changelog should also be updated.

File: passive/JavaDisclosure.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+//Passive scan for Java error messages containing sensitive information (CWE-209)
+
+function scan(ps, msg, src) {
+    var alertRisk = 2
+    var alertConfidence = 3
+    var alertTitle = 'Java stack trace disclosure (or similar) - investigation required (script)'
```
Comment (kingthorin) [inline passive/JavaDisclosure.js:6]: That's quite a huge title

File: passive/JavaDisclosure.js (hunk 1 above)
Comment (ChieftainY2k) [reply passive/JavaDisclosure.js]: Fixed 👍

**Comment by thc202** [comment]
There's a conflict with the changelog, you need to rebase or merge.

**Comment by ChieftainY2k** [comment]
> There's a conflict with the changelog, you need to rebase or merge.

Merged.

**Review by kingthorin — APPROVED** [review]

**Comment by thc202** [comment]
Could you include latest changes? Should make the build green.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by ChieftainY2k** [comment]
Thanks guys!

**Pull request merged**